plugins/repository-azure/src/test/java/org/elasticsearch/repositories/azure/AzureStorageCleanupThirdPartyTests.java

```diff
@@ -71,5 +81,32 @@ protected void createRepository(String repoName) {
                 .put("base_path", System.getProperty("test.azure.base"))
             ).get();
         assertThat(putRepositoryResponse.isAcknowledged(), equalTo(true));
+        if (Strings.hasText(System.getProperty("test.azure.sas_token"))) {
+            ensureSasTokenHasMinimalPermissions();
+        }
+    }
+
+    private void ensureSasTokenHasMinimalPermissions() {
+        final BlobStoreRepository repository = getRepository();
+        final PlainActionFuture<Void> future = PlainActionFuture.newFuture();
+        repository.threadPool().generic().execute(ActionRunnable.wrap(future, l -> {
```

Comment:
Admitttedly, this is a pretty brute-force approach to ensuring that the tokem doesn't exceed necessary permissions but given the impact a mistake here could have I'd rather do this than trying to hackily parse the permissions out of the token (not sure that's even safely possible at the granularity we need).